sources/packages/api/src/route-controllers/confirmation-of-enrollment/confirmation-of-enrollment.controller.ts

```diff
@@ -250,32 +291,47 @@ export class ConfirmationOfEnrollmentController {
    */
   @HasLocationAccess("locationId")
   @Patch(
-    ":locationId/confirmation-of-enrollment/application/:applicationId/deny",
+    ":locationId/confirmation-of-enrollment/disbursement/:disbursementScheduleId/deny",
   )
   async denyConfirmationOfEnrollment(
     @Param("locationId") locationId: number,
-    @Param("applicationId") applicationId: number,
+    @Param("disbursementScheduleId") disbursementScheduleId: number,
     @Body() payload: DenyConfirmationOfEnrollmentDto,
+    @UserToken() userToken: IUserToken,
   ): Promise<void> {
-    try {
-      const application = await this.applicationService.setDeniedReasonForCOE(
-        applicationId,
+    if (
+      payload.coeDenyReasonId === COE_DENIED_REASON_OTHER_ID &&
+      !payload.otherReasonDesc
+    ) {
+      throw new UnprocessableEntityException(
+        "Other is selected as COE reason, specify the reason for the COE denial.",
+      );
+    }
+    const disbursementSchedule =
+      await this.disbursementScheduleService.getDisbursementAndApplicationSummary(
         locationId,
-        payload.coeDenyReasonId,
-        payload.otherReasonDesc,
+        disbursementScheduleId,
       );
-      if (application.assessmentWorkflowId) {
-        await this.workflow.deleteApplicationAssessment(
-          application.assessmentWorkflowId,
-        );
-      }
-    } catch (error) {
-      if (error.name === COE_REQUEST_NOT_FOUND_ERROR) {
-        throw new UnprocessableEntityException(error.message);
-      }
 
-      throw new InternalServerErrorException(
-        "Error while denying a Confirmation Of Enrollment (COE).",
+    if (!disbursementSchedule) {
+      throw new NotFoundException(
+        "Unable to find a COE which could be completed.",
+      );
+    }
+    await this.disbursementScheduleService.updateCOEToDeny(
+      disbursementSchedule.application.id,
+      userToken.userId,
+      payload.coeDenyReasonId,
+      payload.otherReasonDesc,
+    );
+
+    if (
+      disbursementSchedule.application.applicationStatus ===
+        ApplicationStatus.enrollment &&
+      disbursementSchedule.application.assessmentWorkflowId
+    ) {
+      await this.workflow.deleteApplicationAssessment(
```

Comment:
I believe that we will not be deleting the workflow on the second COE because it is supposed to be done at this point and the application completed, right?

Reply:
Yes @andrewsignori-aot. When first COE is complete(approved) the workflow is completed. So if the second COE is rejected, I think we do not and cannot delete the workflow. Correct me If I am missing something.